cnd/src/network/comit.rs

```diff
@@ -254,6 +211,49 @@ impl Comit {
         // We trust in libp2p to poll us.
         Poll::Pending
     }
+
+    fn remote_data_insert<T>(&mut self, shared_swap_id: SharedSwapId, value: T)
+    where
+        RemoteData: Set<T>,
+    {
+        let mut remote_data = self
+            .remote_data
+            .get(&shared_swap_id)
+            .cloned()
+            .unwrap_or_default();
+        remote_data.set(value);
+        self.remote_data.insert(shared_swap_id, remote_data);
+    }
+
+    fn finalize(&mut self, peer: PeerId, shared_swap_id: SharedSwapId) {
```

Comment:
:thinking: This function is pretty tricky to understand for me.

Reply:
I agree, I also am not a fan of this style of code. I intentionally did not put time into simplifying this for this PR. Will do it when I get some time allocated to clean up this module, adding it to my TODO list.